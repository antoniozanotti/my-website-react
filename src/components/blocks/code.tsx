export default function Code() {
  const colorConst = "text-accent-2 dark:text-dark-accent-2";
  const colorIdentifier = "text-accent-1 dark:text-dark-accent-1";
  const colorObject = "text-accent dark:text-dark-accent";
  const colorString = "text-primary-2 dark:text-dark-primary-2";
  return (
    <code>
      <span className={colorConst}>const</span>{" "}
      <span className={colorIdentifier}>birthday</span> ={" "}
      <span className={colorConst}>new</span>{" "}
      <span className={colorObject}>Date</span>(
      <span className={colorString}>{"\"1990-04-18\""}</span>);
      <br />
      <span className={colorConst}>const</span>{" "}
      <span className={colorIdentifier}>location</span> ={" "}
      <span className={colorString}>{"\"Florian√≥polis, Brazil\""}</span>;<br />
      <span className={colorConst}>const</span>{" "}
      <span className={colorIdentifier}>pronouns</span> = [
      <span className={colorString}>{"\"he\""}</span>,{" "}
      <span className={colorString}>{"\"him\""}</span>];
      <br />
      <span className={colorConst}>const</span>{" "}
      <span className={colorIdentifier}>degree</span> ={" "}
      <span className={colorString}>
        {"\"Internet Systems Technologist (FEEVALE, 2011)\""}
      </span>
      ;<br />
      <span className={colorConst}>const</span>{" "}
      <span className={colorIdentifier}>secondDegree</span> ={" "}
      <span className={colorString}>{"\"Bachelor in Sociology\""}</span>;<br />
      <span className={colorConst}>const</span>{" "}
      <span className={colorIdentifier}>values</span> = [
      <span className={colorString}>{"\"ethic\""}</span>,{" "}
      <span className={colorString}>{"\"love\""}</span>,{" "}
      <span className={colorString}>{"\"freedom\""}</span>];
    </code>
  );
}
