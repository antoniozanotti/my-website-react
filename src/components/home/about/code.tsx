import { useTranslations } from "next-intl";

export default function Code() {
  const tAbout = useTranslations('About');
  const t = useTranslations('');
  const colorConst = "text-accent-2 dark:text-dark-accent-2";
  const colorIdentifier = "text-accent-1 dark:text-dark-accent-1";
  const colorObject = "text-accent dark:text-dark-accent";
  const colorString = "text-primary-2 dark:text-dark-primary-2";
  return (
    <code>
      <span className={colorConst}>const</span>{" "}
      <span className={colorIdentifier}>{tAbout('varBirthday')}</span> ={" "}
      <span className={colorConst}>new</span>{" "}
      <span className={colorObject}>Date</span>(
      <span className={colorString}>{"\"1990-04-18\""}</span>);
      <br />
      <span className={colorConst}>const</span>{" "}
      <span className={colorIdentifier}>{tAbout('varLocation')}</span> ={" "}
      <span className={colorString}>{"\"Florian√≥polis, "+t("Brazil")+"\""}</span>;<br />
      <span className={colorConst}>const</span>{" "}
      <span className={colorIdentifier}>{tAbout('varPronouns')}</span> = [
      <span className={colorString}>{"\""+tAbout("pronouns.he")+"\""}</span>,{" "}
      <span className={colorString}>{"\""+tAbout("pronouns.him")+"\""}</span>];
      <br />
      <span className={colorConst}>const</span>{" "}
      <span className={colorIdentifier}>{tAbout('varDegree')}</span> ={" "}
      <span className={colorString}>
        {"\""+tAbout("degree")+"\""}
      </span>
      ;<br />
      <span className={colorConst}>const</span>{" "}
      <span className={colorIdentifier}>{tAbout('varValues')}</span> = [
      <span className={colorString}>{"\""+tAbout("values.ethic")+"\""}</span>,{" "}
      <span className={colorString}>{"\""+tAbout("values.love")+"\""}</span>,{" "}
      <span className={colorString}>{"\""+tAbout("values.freedom")+"\""}</span>];
    </code>
  );
}
