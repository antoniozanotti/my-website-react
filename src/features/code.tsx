import React from 'react';
import { ReactNode } from 'react';

export interface CodeProps {
  
}

export const Code = ({
  
}: CodeProps) => {
  return (
    <pre>
      const birthday = new Date(”1990-04-18”); const location =
      “Florianópolis, Brazil”; const pronouns = [“he”, “him”]; const
      degree = “Internet Systems Technologist (FEEVALE, 2011)”; const
      secondDegree = “Bachelor in Sociology”; const values = [“ethic”,
      “love”, “freedom”];
    </pre>
  );
};