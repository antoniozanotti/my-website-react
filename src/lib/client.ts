import { ApolloClient, HttpLink, InMemoryCache } from "@apollo/client";
import { registerApolloClient } from "@apollo/experimental-nextjs-app-support/rsc";

const TOKEN = process.env.CONTENTFUL_ACCESS_TOKEN;
const SPACE = process.env.CONTENTFUL_SPACE_ID;
const ENVIRONMENT = process.env.CONTENTFUL_ENVIRONMENT_ID;
const URL = `https://graphql.contentful.com/content/v1/spaces/${SPACE}/environments/${ENVIRONMENT}`;

export const { getClient } = registerApolloClient(() => {
  return new ApolloClient({
    cache: new InMemoryCache(),
    link: new HttpLink({
      uri: URL,
      headers: {
        Authorization: `Bearer ${TOKEN}`,
      },
    }),
  });
});
