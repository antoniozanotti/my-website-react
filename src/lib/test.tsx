import React, { ReactElement } from "react";
import {
  render,
  RenderOptions,
  RenderResult,
  Screen,
} from "@testing-library/react";
import { Theme } from "@/domain/theme/theme/theme";
import { Providers } from "../app/providers";
import messages from "@/messages/en.json";

function AllTheProviders({ children }: { children: React.ReactNode }) {
  return (
    <Providers locale="en" messages={messages}>
      <Theme>
        <div data-testid="container">{children}</div>
      </Theme>
    </Providers>
  );
}

function customRender(
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
): RenderResult {
  return render(ui, { wrapper: AllTheProviders, ...options });
}

function firstComponent(screen: Screen) {
  return screen.getByTestId("container").firstChild;
}

export * from "@testing-library/react";
export * from "@testing-library/user-event";
export { customRender as render, firstComponent };
import "@testing-library/jest-dom";

jest.mock("next/navigation", () => {
  return {
    useRouter() {
      return { push: () => {} };
    },
    useParams: () => ({}),
  };
});
