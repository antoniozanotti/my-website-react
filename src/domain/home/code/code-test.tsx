import { test, expect } from '@playwright/experimental-ct-react';
import { Code } from "./code";

test.describe("Code", () => {
  test("should render code", async ({ mount }) => {
    const component = await mount(<Code />);
    await expect(component.locator(":scope:is(code)")).toBeVisible();
  });
});
