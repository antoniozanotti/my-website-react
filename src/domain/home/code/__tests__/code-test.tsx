import { screen, render } from "@/lib/test";
import { Code } from "../code";

describe("Code", () => {
  it("should render code", async () => {
    const { container } = render(<Code />);
    expect(screen.getByRole("code")).toBeVisible();
    expect(container).toMatchSnapshot();
  });
});
