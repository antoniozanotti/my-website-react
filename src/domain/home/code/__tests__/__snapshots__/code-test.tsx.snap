// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Code should render code 1`] = `
<div>
  <div
    class="dark"
  >
    <div
      class="className bg-dark-2 dark:bg-dark-dark-2 text-primary dark:text-dark-primary text-c3 lg:text-c4"
    >
      <div
        data-testid="container"
      >
        <code>
          <span
            class="text-accent-1 dark:text-dark-accent-1"
          >
            const
          </span>
           
          <span
            class="text-accent-1 dark:text-dark-accent-1"
          >
            birthday
          </span>
           =
           
          <span
            class="text-accent-1 dark:text-dark-accent-1"
          >
            new
          </span>
           
          <span
            class="text-accent dark:text-dark-accent"
          >
            Date
          </span>
          (
          <span
            class="text-primary-2 dark:text-dark-primary-2"
          >
            "1990-04-18"
          </span>
          );
          <br />
          <span
            class="text-accent-1 dark:text-dark-accent-1"
          >
            const
          </span>
           
          <span
            class="text-accent-1 dark:text-dark-accent-1"
          >
            location
          </span>
           =
           
          <span
            class="text-primary-2 dark:text-dark-primary-2"
          >
            "Florian√≥polis, Brazil"
          </span>
          ;
          <br />
          <span
            class="text-accent-1 dark:text-dark-accent-1"
          >
            const
          </span>
           
          <span
            class="text-accent-1 dark:text-dark-accent-1"
          >
            pronouns
          </span>
           = [
          <span
            class="text-primary-2 dark:text-dark-primary-2"
          >
            "he"
          </span>
          ,
           
          <span
            class="text-primary-2 dark:text-dark-primary-2"
          >
            "him"
          </span>
          ];
          <br />
          <span
            class="text-accent-1 dark:text-dark-accent-1"
          >
            const
          </span>
           
          <span
            class="text-accent-1 dark:text-dark-accent-1"
          >
            degree
          </span>
           =
           
          <span
            class="text-primary-2 dark:text-dark-primary-2"
          >
            "Internet Systems Technologist (FEEVALE, 2010)"
          </span>
          ;
          <br />
          <span
            class="text-accent-1 dark:text-dark-accent-1"
          >
            const
          </span>
           
          <span
            class="text-accent-1 dark:text-dark-accent-1"
          >
            values
          </span>
           = [
          <span
            class="text-primary-2 dark:text-dark-primary-2"
          >
            "ethic"
          </span>
          ,
           
          <span
            class="text-primary-2 dark:text-dark-primary-2"
          >
            "love"
          </span>
          ,
           
          <span
            class="text-primary-2 dark:text-dark-primary-2"
          >
            "freedom"
          </span>
          ];
        </code>
      </div>
    </div>
  </div>
</div>
`;
